app/assets/v2/js/grants-ingest-missing-contributions.js

```diff
@@ -0,0 +1,150 @@
+        }
+      }
+
+      if (Object.keys(this.errors).length) {
+        return false; // there are errors the user must correct
+      }
+
+      return {
+        txHash: isValidTxHash ? txHash : '',
+        // getAddress returns checksum address required by web3py, and throws if address is invalid
+        userAddress: isValidAddress ? ethers.utils.getAddress(userAddress) : ''
+      };
+    },
+
+    async ingest(event) {
+      try {
+        event.preventDefault();
+
+        // Return if form is not valid
+        const formParams = this.checkForm();
+
+        if (!formParams) {
+          return;
+        }
+
+        // Send POST requests to ingest contributions
+        const { txHash, userAddress } = formParams;
+        const csrfmiddlewaretoken = document.querySelector('[name=csrfmiddlewaretoken]').value;
+        const url = '/grants/ingest';
+        const headers = { 'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8' };
+        const payload = {
+          csrfmiddlewaretoken,
+          txHash,
+          userAddress,
+          network: document.web3network || 'mainnet'
+        };
+        const postParams = {
+          method: 'POST',
+          headers,
+          body: new URLSearchParams(payload)
+        };
+
+        // Send saveSubscription request
+        const res = await fetch(url, postParams);
+        const json = await res.json();
+        
+        // Notify user of success status, and clear form if successful
+        console.log('ingestion response: ', json);
+        if (!json.success) {
+          console.log('ingestion failed');
+          this.submitted = false;
+          throw new Error('Your transactions could not be processed, please try again');
+        } else {
+          console.log('ingestion successful');
+          _alert('Your contributions have been added successfully!', 'success');
+          this.resetForm();
+        }
+      } catch (e) {
+        this.handleError(e);
+      }
+    },
+
+    resetForm() {
+      this.form.txHash = undefined;
+      this.form.userAddress = undefined;
+      this.errors = {};
+      this.submitted = false;
+    },
+
+    handleError(err) {
+      console.error(err); // eslint-disable-line no-console
+      let message = 'There was an error';
+
+      if (err.message)
+        message = err.message;
+      else if (err.msg)
+        message = err.msg;
+      else if (typeof err === 'string')
+        message = err;
+
+      _alert(message, 'error');
+      this.submitted = false;
+    }
+  },
+
+  watch: {
+    deep: true,
+    form: {
+      deep: true,
+      handler(newVal, oldVal) {
+        this.checkForm();
+        this.submitted = false;
+        this.errors = {};
+      }
+    }
+  }
+});
+
+if (document.getElementById('gc-grants-ingest-contributions')) {
+
+  appIngestContributions = new Vue({
+    delimiters: [ '[[', ']]' ],
+    el: '#gc-grants-ingest-contributions'
+  });
+}
```

Comment:
can we throw this into `/grants/ingest-missing-contributions.js`

Reply:
Do you mean move this whole JS file from `app/assets/v2/js/grants-ingest-missing-contributions.js` into `app/grants/ingest-missing-contributions.js`?

If so, sure. One question though—is there some sort of guidelines we're using to decide where to put JS/Vue files? It seems most JS files (including grants related ones) are located in `app/assets/v2/js/` which is why I put it there

Reply:
well it's always easier to group files together unless it's to be reused !
Folks do end up creating multiple files and I end up having to port it when I get the time 😅

Reply:
@mds1 ^

Reply:
Moved the JS file into `app/assets/v2/js/grants/ingest-missing-contributions.js` in 191d3f2